pre/remap_restart/remap_questions.py

```diff
@@ -13,383 +13,374 @@
 
 def ask_questions():
 
+   choices_bc_ops     = ['NL3', 'ICA', 'GM4', 'Other']
+   choices_bc_other   = ['v06']
+
+   choices_omodel     = ['data', 'MOM5', 'MOM6']
+
+   choices_ogrid_data = ['360x180   (Reynolds)','1440x720  (MERRA-2)','2880x1440 (OSTIA)','CS  (same as atmosphere OSTIA cubed-sphere grid)']
+
+   choices_ogrid_cpld = ['72x36', '360x200', '720x410', '1440x1080']
+
    questions = [
         {
             "type": "confirm",
             "name": "input:shared:MERRA-2",
-            "message": "Would you like to remap archived MERRA-2 restarts?",
+            "message": "Remap archived MERRA-2 restarts?\n",
             "default": False,
         },
         {
             "type": "path",
             "name": "input:shared:rst_dir",
-            "message": "Enter the directory containing restart files to be remapped:",
+            "message": "Enter input directory with restart files to be remapped:\n",
             "when": lambda x: not x['input:shared:MERRA-2'],
         },
         {
             "type": "text",
             "name": "input:shared:yyyymmddhh",
-            "message": "From what restart date/time would you like to remap? (must be 10 digits: yyyymmddhh)",
+            "message": "Enter restart date/time:  (Must be 10 digits: yyyymmddhh.)\n",
             "validate": lambda text: len(text)==10 ,
+            "when": lambda x: not x['input:shared:MERRA-2'] and fvcore_time(x),
+        },
+        {
+            "type": "text",
+            "name": "input:shared:yyyymmddhh",
+            "message": "Enter restart date:  (Must be 8 digits: yyyymmdd; hour=21z.)\n",
+            "validate": lambda text: len(text)==8 ,
+            "when": lambda x: x['input:shared:MERRA-2'],
+        },
+        {
+            "type": "path",
+            "name": "output:shared:out_dir",
+            "message": "Enter output directory for new restarts:\n"
         },
 
+        # dummy (invisible) question to run function that initializes MERRA-2 config
         {
             "type": "path",
             "name": "output:shared:out_dir",
-            "message": "Enter the directory for new restarts:\n"
+            "message": "init merra2\n",
+            # always return false, so question never shows but changes x
+            "when": lambda x: init_merra2(x),
         },
 
         {
             "type": "text",
             "name": "input:shared:agrid",
-            "message": "Enter input atmospheric grid: \n C12   C180  C1000  C270\n C24   C360  C1440  C540\n C48   C500  C2880  C1080\n C90   C720  C5760  C2160  C1536\n ",
-            "default": 'C360',
+            "message": f'''Enter atmospheric grid of input restarts:
+    C12   C180  C1000  C270
+    C24   C360  C1440  C540
+    C48   C500  C2880  C1080
+    C90   C720  C5760  C2160  C1536\n''',
+            "validate": lambda text : text in ['C12','C180','C1000','C270','C24','C360','C1440','C540','C48','C500','C2880','C1080','C90','C720','C5760','C2160','C1536'],
```

Comment:
Can we also put multi-line "message" strings into variables (like, e.g., `choices_bc_ops`)?  This particular block is needed twice, and it would be better to have it in just one place.  Likewise for the list in "validate".

Reply:
Yes, that makes the code cleaner